pkg/apps/gitops.go

```diff
@@ -148,3 +150,68 @@ func (o *GitOpsOptions) DeleteApp(app string, alias string) error {
 	log.Infof("Delete app via Pull Request %s\n", info.PullRequest.URL)
 	return nil
 }
+
+// GetApps retrieves all the apps information for the given appNames from the repository and / or the CRD API
+func (o *GitOpsOptions) GetApps(appNames map[string]bool, expandFn func([]string) (*v1.AppList, error)) (*v1.AppList, error) {
+
+	options := environments.EnvironmentPullRequestOptions{
+		ConfigGitFn:   o.ConfigureGitFn,
+		Gitter:        o.Gitter,
+		ModifyChartFn: nil,
+		GitProvider:   o.GitProvider,
+	}
+
+	dir, _, _, _, err := options.PullEnvironmentRepo(o.DevEnv, o.EnvironmentsDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "couldn't pull the environment repository from %s", o.DevEnv.Name)
+	}
+
+	envDir := filepath.Join(dir, helm.DefaultEnvironmentChartDir)
+	if err != nil {
+		return nil, err
+	}
+	exists, err := util.DirExists(envDir)
+	if err != nil {
+		return nil, err
+	}
+
+	if !exists {
+		envDir = dir
+	}
+
+	requirementsFile, err := ioutil.ReadFile(filepath.Join(envDir, helm.RequirementsFileName))
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't read the environment's requirements.yaml file")
+	}
+	reqs := helm.Requirements{}
+	err = yaml.Unmarshal(requirementsFile, &reqs)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't unmarshal the environment's requirements.yaml file")
+	}
+
+	appsList := v1.AppList{}
+	for _, d := range reqs.Dependencies {
+		if appNames[d.Name] == true || len(appNames) == 0 {
+			//Make sure we only consider apps
+			if match, _ := regexp.MatchString("jx-app-.*", d.Name); match {
```

Comment:
Sorry, didn't spot this before, but we can't do this as there is no requirement for an app to be called `jx-app-XXX`, that's just a convention. Every requirement except `jenkins-x-platform` is an app right now.